src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs

```diff
@@ -86,8 +85,42 @@ protected void OnConfigurationChanged(List<ReplicationDestination> destinations)
                 foreach (var dest in destinations)
                 {
                     var outgoingDocumentReplication = new OutgoingDocumentReplication(_database, dest);
-                    initializationTasks.Add(outgoingDocumentReplication.InitializeAsync());
-                    _outgoingReplications.Add(outgoingDocumentReplication);
+                    initializationTasks.Add(
+						outgoingDocumentReplication.InitializeAsync()
+												   .ContinueWith(t =>
```

Comment:
Didn't even try to read it. This is too complex.
Why is this async?
Why are you using ContinueWith?